atc/db/resource_type.go

```diff
@@ -83,8 +81,8 @@ func (resourceTypes ResourceTypes) Filter(checkable Checkable) ResourceTypes {
 	}
 }
 
-func (resourceTypes ResourceTypes) Deserialize() atc.VersionedResourceTypes {
-	var versionedResourceTypes atc.VersionedResourceTypes
+func (resourceTypes ResourceTypes) Deserialize() atc.ResourceTypes {
+	var atcResourceTypes atc.ResourceTypes
```

Comment:
If we init it with an empty array, then caller will not need to check nil.

Reply:
I'm not exactly sure what you mean here, I don't think we have anywhere that requires us to check if the resource types is nil or not?